x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/vectors/30_dense_vector_script_access.yml

```diff
@@ -0,0 +1,142 @@
+---
+"Access to values of dense_vector in script":
+  - skip:
+      features: headers
+      version: " - 7.12.99"
+      reason: "Access to values of dense_vector in script was added in 7.13"
+  - do:
+      indices.create:
+        index: test-index
+        body:
+          mappings:
+            properties:
+              v:
+                type: dense_vector
+                dims: 3
+
+  - do:
+      bulk:
+        index: test-index
+        refresh: true
+        body:
+          - '{"index": {"_id": "1"}}'
+          - '{"v": [1, 1, 1]}'
+          - '{"index": {"_id": "2"}}'
+          - '{"v": [1, 1, 2]}'
+          - '{"index": {"_id": "3"}}'
+          - '{"v": [1, 1, 3]}'
+          - '{"index": {"_id": "missing_vector"}}'
+          - '{}'
+
+  # check getVectorValue() API
+  - do:
```

Comment:
This test coverage looks good! Since unit tests are generally easier to work with than REST tests, I wondered if there was a way perform some of the same checks as unit tests?

Reply:
@jtibshirani Thanks for the feedback. I could not find any good examples of doing unit tests with script doc values.  Unit tests that we have they mock scripts contexts and mock what script returns, which kind of defeats the purpose of testing what `getVectorValue()` and `getMagnitude()` returns.

I am happy to redesign tests as unit tests, if you know any examples I can follow.

Reply:
It seems like there are some simple cases where we just want to check `getVectorValue` and `getMagnitude` return the right value or error appropriately. These could be covered in a test like `DenseVectorScriptDocValuesTests`. A similar test would be `ScriptDocValuesGeoPointsTests`.